frontend/src/components/object/ObjectTable.vue

```diff
@@ -178,13 +199,11 @@ watch( selectedObjects, () => {
       >
         <template #body="{ data }">
           <ShareObjectButton
-            v-if="permissionStore.isObjectActionAllowed(
-              data.id, getUserId, Permissions.READ, props.bucketId as string)"
             :id="data.id"
           />
           <DownloadObjectButton
             v-if="permissionStore.isObjectActionAllowed(
-              data.id, getUserId, Permissions.READ, props.bucketId as string)"
+              data.id, getUserId, Permissions.READ, props.bucketId as string) || data.public"
```

Comment:
It's better to evaluate obj.public first to leverage short circuiting; that way we can save some computation if it's already public and avoid needing to hit the permissionStore.